service/src/main/java/org/apache/celeborn/server/common/service/config/DynamicConfig.java

```diff
@@ -42,7 +42,7 @@
  */
 public abstract class DynamicConfig {
   private static final Logger LOG = LoggerFactory.getLogger(DynamicConfig.class);
-  protected Map<String, String> configs = new HashMap<>();
+  protected volatile Map<String, String> configs = new HashMap<>();
```

Comment:
Why not use ConcurrentHashMap?

Reply:
This map won't be updated by multiple threads concurrently.